validator/keymanager/wallet.go

```diff
@@ -0,0 +1,110 @@
+package keymanager
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/bytesutil"
+	e2wallet "github.com/wealdtech/go-eth2-wallet"
+	e2wtypes "github.com/wealdtech/go-eth2-wallet-types"
+)
+
+type walletOpts struct {
+	Accounts    []string `json:"accounts"`
+	Passphrases []string `json:"passphrases"`
+}
+
+var walletOptsHelp = `The wallet key manager stores keys in a local encrypted store.  The options are:
+  - accounts This is a list of account specifiers.  An account specifier is of
+    the form <wallet name>/[account name],  where the account name can be a
+	regular expression.  If the account specifier is just <wallet name> all
+	accounts in that wallet will be used.  Multiple account specifiers can be
+	supplied if required.
+  - passphrase This is the passphrase used to encrypt the accounts when they
+    were created.  Multiple passphrases can be supplied if required.
+  {
+    "accounts":    ["Validators/Account.*"], // Use all accounts in the 'Validators' wallet starting with 'Account'
+    "passphrases": ["secret1","secret2"]     // Use the passphrases 'secret1' and 'secret2' to decrypt accounts
+  }`
+
+// NewWallet creates a key manager populated with the keys from a wallet at the given path.
+func NewWallet(input string) (KeyManager, string, error) {
+	opts := &walletOpts{}
+	err := json.Unmarshal([]byte(input), opts)
+	if err != nil {
+		return nil, walletOptsHelp, err
+	}
+
+	if len(opts.Accounts) == 0 {
+		return nil, walletOptsHelp, errors.New("at least one account specifier is required")
+	}
+
+	if len(opts.Passphrases) == 0 {
+		return nil, walletOptsHelp, errors.New("at least one passphrase is required to decrypt accounts")
+	}
+
+	km := &Wallet{
+		accounts: make(map[[48]byte]e2wtypes.Account),
+	}
+
+	for _, path := range opts.Accounts {
+		parts := strings.Split(path, "/")
+		if len(parts[0]) == 0 {
+			return nil, walletOptsHelp, fmt.Errorf("did not understand account specifier %q", path)
+		}
+		wallet, err := e2wallet.OpenWallet(parts[0])
+		if err != nil {
+			return nil, walletOptsHelp, err
+		}
+		accountSpecifier := "^.*$"
+		if len(parts) > 1 && len(parts[1]) > 0 {
+			accountSpecifier = fmt.Sprintf("^%s$", parts[1])
+		}
+		re := regexp.MustCompile(accountSpecifier)
```

Comment:
do we need to use regex here. Its not very performant in go, if you could do a direct substring comparison, maybe that will be better ?

Reply:
I haven't seen any performance issues running with a decent amount of validators (4096).  Pre-compilation of the regex helps a lot here compared to running a match in `strings`.  Did you see a particular issue?

Reason for the regex is that we're not guaranteed to only be running with simple distributions of keys.  For example we could have a sharded based system where all keys ending in '0' are looked after by one validator, all ending in '1' are looked after by another, '2' a different one again, _etc._  Also, I'm sure that people will come up with weird and wonderful naming conventions for their keys so wanted to be as flexible as possible in all situations.

Reply:
That makes sense, I was worried about compiling regex for every key, but it looks like you precompile it before hand once so that should mostly help when using it with multiple keys. This is fine then